examples/nrf/src/bin/channel.rs

```diff
@@ -0,0 +1,45 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use defmt::unwrap;
+use embassy::blocking_mutex::raw::ThreadModeRawMutex;
+use embassy::channel::channel::Channel;
+use embassy::executor::Spawner;
+use embassy::time::{Duration, Timer};
+use embassy_nrf::gpio::{Level, Output, OutputDrive};
+use embassy_nrf::Peripherals;
+
+use defmt_rtt as _; // global logger
+use panic_probe as _;
+
+enum LedState {
+    On,
+    Off,
+}
+
+static CHANNEL: Channel<ThreadModeRawMutex, LedState, 1> = Channel::new();
+
+#[embassy::task]
+async fn my_task() {
```

Comment:
I realise that this example doesn't need it, but I think it'd be great to show the Sender being passed in as a param to the task as that is commonly how it will be passed around.

Reply:
> that is commonly how it will be passed around.

I'm not so sure! Using the `static` directly is much easier and might save a tiny bit of RAM (no need to pass around a pointer if you can just go and get the static directly).

One advantage for using Sender/Receiver is that you can use NoopMutex instead. I'll add another example!

Reply:
My tasks are often in their own modules. Here's a declaration of one for processing analog inputs for the existing MPSC:

```rust
#[embassy::task(pool_size = 1)]
pub async fn main_task(
    mut p: AnalogInputPeripherals,
    line_flow_signal_0: &'static Signal<line_monitor::LineFlow>,
    line_flow_signal_1: &'static Signal<line_monitor::LineFlow>,
    sender: Sender<'static, NoopRawMutex, state::Command, MAX_COMMANDS_PENDING>,
)
```

While the signal is static, I still find it useful to pass in as a param as it feels clear as to what the task's requirements are.

Reply:
Added another example